Added comments tab of event details modal

SimoneSchmidt opened the pull request:
UI components and functionality was added:

- In the event table, clicking on the actions for 'open event details', 'view comments' and 'open asset details' now opens a modal with (so far mostly empty) tabs for the different event details.
- Comments tab in the event details modal was added. Aside from role and user management, functionality is finished and should work as expected.

SimoneSchmidt committed:
added on-click functions for event actions

SimoneSchmidt committed:
this commit doesn't do anything useful

SimoneSchmidt committed:
Merge branch 'admin-ui-picard' of https://github.com/TIK-NFL/opencast into admin-ui-picard

SimoneSchmidt committed:
Merge branch 'admin-ui-picard' into admin-ui-picard_develop

SimoneSchmidt committed:
added modal and wizard for event details dialog

SimoneSchmidt committed:
started implementing comments tab for event details

SimoneSchmidt committed:
Finished UI components for Comments Tab of Event Details, some of the checks when saving a comment and background actions still missing

SimoneSchmidt committed:
Merge branch 'admin-ui-picard' into admin-ui-picard_develop

SimoneSchmidt committed:
Merge branch 'admin-ui-picard' into admin-ui-picard_develop

SimoneSchmidt committed:
Comments Tab of Event Details

isabellakutger commented on modules/admin-ui-picard/app/src/components/events/partials/modals/EventDetails.js:
```diff
@@ -0,0 +1,206 @@
+                <div className="full-col">
+                    <div className="obj tbl-details">
+                        <header className="no-expand">{t(header)}</header>
+                        {/* Table view containing input fields for metadata */}
```
Remove this comment

isabellakutger commented on modules/admin-ui-picard/app/src/components/events/partials/modals/EventDetailsCommentsTab.js:
```diff
@@ -0,0 +1,250 @@
+    return (
+        <div className="modal-content">
+            <div className="modal-body">
+                { /* <div data-admin-ng-notifications="" context="events-access"/> */}
```
Here you can use the Notification component

isabellakutger commented on modules/admin-ui-picard/app/src/components/events/partials/modals/EventDetailsModal.js:
```diff
@@ -0,0 +1,37 @@
+
+
+/**
+ * This component renders the modal for adding new resources
```
Adjust comment

isabellakutger commented on modules/admin-ui-picard/app/src/components/events/partials/modals/EventDetailsModal.js:
```diff
@@ -0,0 +1,37 @@
+                    <header>
+                        <a className="fa fa-times close-modal" onClick={() => close()}/>
+                        <h2>{t('EVENTS.EVENTS.DETAILS.HEADER',
+                            { resourceId: "Beispiel Event ID" /*todo: find real value*/} ) /*Event details - {{resourceId}}*/} </h2>
```
Translate placeholder to english

isabellakutger commented on modules/admin-ui-picard/app/src/thunks/eventDetailsThunks.js:
```diff
@@ -0,0 +1,85 @@
+        dispatch(saveCommentInProgress());
+
+        const commentSaved = await axios.post(`admin-ng/event/${eventId}/comment`,
+            `text=${commentText}&reason=${commentReason}`);
```
use axios `config` parameter with `param` here. You can find an example in seriesThunk.js (fetchSeries)

isabellakutger commented on modules/admin-ui-picard/app/src/thunks/eventDetailsThunks.js:
```diff
@@ -0,0 +1,85 @@
+
+export const deleteComment = (eventId, commentId) => async (dispatch) => {
+    try {
+        const commentDeleted = await axios.delete(`admin-ng/event/${eventId}/comment/${commentId}`);
```
See comment on `saveComment`

isabellakutger commented on modules/admin-ui-picard/app/src/thunks/eventDetailsThunks.js:
```diff
@@ -0,0 +1,85 @@
+    }
+}
+
+export const deleteComment = (eventId, commentId) => async (dispatch) => {
```
you don't need `dispatch` here, because you not working with redux state and actions here.
If you want to fire a notification then you would need `dispatch`

isabellakutger commented on modules/admin-ui-picard/app/src/thunks/eventDetailsThunks.js:
```diff
@@ -0,0 +1,85 @@
+
+        const commentReply = await axios.post(`admin-ng/event/${eventId}/comment/${commentId}/reply`,
+            `text=${replyText}&resolved=${commentResolved}`);
+
```
see comment on `saveComment`

isabellakutger commented on modules/admin-ui-picard/app/src/thunks/eventDetailsThunks.js:
```diff
@@ -0,0 +1,85 @@
+
+export const deleteCommentReply = (eventId, commentId, replyId) => async (dispatch) => {
+    try {
+        const commentReplyDeleted = await axios.delete(`admin-ng/event/${eventId}/comment/${commentId}/${replyId}`);
```
see comment on `saveComment`

isabellakutger commented on modules/admin-ui-picard/app/src/thunks/eventDetailsThunks.js:
```diff
@@ -0,0 +1,85 @@
+    }
+}
+
+export const deleteCommentReply = (eventId, commentId, replyId) => async (dispatch) => {
```
Here again no `dispatch` needed

isabellakutger reviewed (changes requested):
no major issues, only some small changes needed

SimoneSchmidt committed:
done with comments tab of event details; fixed requested changes from pull request

isabellakutger commented on modules/admin-ui-picard/app/src/components/events/partials/modals/EventDetailsCommentsTab.js:
```diff
@@ -0,0 +1,251 @@
+    return (
+        <div className="modal-content">
+            <div className="modal-body">
+                <Notifications context="events-access"/>
```
context="not-corner"

isabellakutger commented on modules/admin-ui-picard/app/src/thunks/eventDetailsThunks.js:
```diff
@@ -0,0 +1,102 @@
+export const deleteComment = (eventId, commentId) => async (getState) => {
+    try {
+        const state = getState();
+        let params = getURLParams(state);
```
Not sure you can use this function because its used for the url params needed in the request for resources displayed in the main table view. Maybe take a look what this function does and define the params you need accordingly.

SimoneSchmidt replied:
delete doesn't have any parameters, as far as I can see in the old version of the ui, I will remove the params part again from the delete requests

isabellakutger commented on modules/admin-ui-picard/app/src/thunks/eventDetailsThunks.js:
```diff
@@ -0,0 +1,102 @@
+        let params = getURLParams(state);
+
+        const commentReply = await axios.post(`admin-ng/event/${eventId}/comment/${commentId}/reply`,
+            `text=${replyText}&resolved=${commentResolved}`, { params: params });
```
text and resolved should be defined in params too

isabellakutger commented on modules/admin-ui-picard/app/src/thunks/eventDetailsThunks.js:
```diff
@@ -0,0 +1,102 @@
+        dispatch(saveCommentReplyInProgress());
+
+        const state = getState();
+        let params = getURLParams(state);
```
see comment above

isabellakutger commented on modules/admin-ui-picard/app/src/thunks/eventDetailsThunks.js:
```diff
@@ -0,0 +1,102 @@
+export const deleteCommentReply = (eventId, commentId, replyId) => async (getState) => {
+    try {
+        const state = getState();
+        let params = getURLParams(state);
```
see comment above

SimoneSchmidt committed:
adressed requested changes for pull request

adressed the requested changes for the pull request from 2021-05-03
fixed the context for the Notifications tag in EventDetailsCommentsTab.js
removed the unnecessary url parameters from the deleteComment and deleteCommentReply functions in eventDetailsThunks.js
fixed the request parameters and http-headers for saveComment and saveCommentReply functions in eventDetailsThunks.js
